### ISSUE TITLE
What is meant by "a set of web pages"?

### ISSUE THREAD

**Issue by guyhickling** [w3c/wcag#1367]
The phrase "a set of web pages" is used in various SCs, but the glossary definition of it is extremely vague. Just "collection of web pages that share a common purpose....". That is not much clearer than the original phrase. 

So my question is, does a complete website count as a "a set of web pages"? Most websites are made of pages "that share a common purpose" in that they all have the purpose of telling people about a particular organisation, or else provide information on a particular subject. So a website would appear to fit the definition.

The answer to this question is now more important than it has been, since the new Findable Help SC (3.2.6) expressly requires that help info is provided on each page of "any set of Web pages". That might not be appropriate for all pages on a website, yet the SC would require it to be done nevertheless.

**Comment by alastc** [comment]
Hi Guy,

> So my question is, does a complete website count as a "a set of web pages"?

Yes, if the website is of one common purpose, a website could fit the definition.

However, many larger websites would not, and the definition is useful to allow them to have separate navigations etc. For example, the products section might have a different navigation from the accounts section. Different navigation menus might be appropriate (SC 3.2.3), and different help mechanisms might be available (3.2.6).

It was used in WCAG 2.0 criteria so it has been around for 12 years now.

**Comment by guyhickling** [comment]
As you say, a website can have several sets of pages in it. So now, in order to test whether any one page passes or fails the SC, we have to be able to say, with certainty, that the page belongs to a particular "set" (if we are going to require the same Help info to be shown on it as on the other pages in a set). 

However the term "set of pages" is too vague to be testable. Who is to say with any certainty, whether a particular page belongs to any particular "set" or not? And if that is not testable, then any SC that relies on that page being in a particular set is not a testable SC. One of the prime requirements for becoming an SC is: "A Success Criterion is a testable statement that will be either true or false when applied to specific Web content."

I'm arguing that this SC does not meet that requirement.

> It was used in four WCAG 2.0 criteria

True, but in all those other SCs there is no requirement for one page to be identified as part of a particular set (a page could even be part of more than one set). Those SCs simply required that if some content showed on multiple pages, then it had to be shown consistently on all those pages where it showed. 

For instance, the Consistent Navigation SC says "Navigational mechanisms that are repeated on multiple Web pages....". It doesn't require that navigation to show on all pages in the set, only that where it does show it must show consistently. 

This new SC is different. It requires that the same content has to shown on all pages in the set, so now we have to be able to identify, with certainly, all the pages in the set.

**Comment by scha14** [comment]
New response approved by the group, same as[ issue 1427](https://github.com/w3c/wcag/issues/1427 ), so closing. 

<hr>

> I don't see how a UI component can be included in the same relative order on each page within a single page web application.
> Perhaps "single page" should be replaced with "multiple page"?

Part of the confusion comes from the original definition of web-page, which is based on the URI, so oddly (to modern thinking) an SPA which does not update the URL is a 'web page', not a set of web pages.

For this SC we have three conditions covered by 2 definitions:

- Web pages, in the traditional sense.
- SPAs with different URIs, covered by the standard definition of web pages and 'set of web pages'.
- SPA without routing (same URI), covered by the new SPA definition.

It is possible in an SPA (same URI style) could move navigation and/or help links around the layout, so the SC is intended to apply in this instance.

Please note we have added a paragraph on SPAs to the understanding document recently:
> A single page Web application (compared to a Web page) shows multiple "pages" or views of content at the same URI. If a web application uses different URIs for each view of the content, that is considered multiple Web pages because the URI changes.

With that update to the understanding document, and this explanation, the group considers this issue addressed, but please re-open if there is something about this issues that you don't consider to be addressed.

**Comment by alastc** [comment]
Hi @scha14, I don't think this is the same issue, so I'll re-open this one.

We discussed it here: https://www.w3.org/2020/12/01-ag-minutes.html#item03
Survey results: https://www.w3.org/2002/09/wbs/35422/findable-help-issues/results#xq9

The crux was this comment: 
> So "Consistent Navigation" doesn't require that navigation be shown on all pages in the set, only that where it does show it must show consistently.
> 
> I think this does put us under a burden of proof to show, either through examples or a definition update, how people can consistently identify whether any page is part of a set.
> 
> If that is too difficult, it might be a case for deferment because in Silver the person claiming conformance defines the process/task/set.

The resolution was to "Update response to issue #1367 and include additional examples in understanding docs, and review in a subsequent meeting."

We need to get someone looking at examples to add.

**Comment by scha14** [comment]
Ah sorry, my bad. That was one of the meetings I missed and thought we had the same response for this one.

**Comment by alastc** [comment]
Hi everyone,

The text of the [success criterion](https://raw.githack.com/w3c/wcag/findable_help_february_2021/understanding/22/findable-help.html) has been significantly updated, and the group believes it resolves the issue so this issue is being closed. There will be another review opened soon.
